driver-core/src/main/com/mongodb/ConnectionString.java

```diff
@@ -674,6 +679,9 @@ private MongoCredential createCredentials(final Map<String, List<String>> option
             if (key.equals("authmechanism")) {
                 mechanism = AuthenticationMechanism.fromMechanismName(value);
             } else if (key.equals("authsource")) {
+                if (userName == null && !isSrvProtocol) {
```

Comment:
This change worries me.  I didn't realize it when I reviewed the original SPEC PR, but this is a backwards breaking change that could affect users.  I wouldn't have notice it had you not included the `!isSrvProtocol` condition.  That condition is not actually specified or tested in the spec tests, so it's possible that other driver would not include that condition in their own implementations.  But it's an important one because Atlas TXT records do typically include an authSource query parameter, so without this condition it would be impossible to create Atlas-based apps that use mongodb+srv connection strings but do not include credentials.  For instance, a monitoring app that just calls the `ping` command.

But if we keep this exception to the rule, then we have introduced an inconsistency: connection strings that specify authSource explicitly work one way, while connection strings that specify authSource indirectly through a TXT record work another way.

Given this, I'd like to revisit this one part of the spec change.  I'm going to open a SPEC ticket with the above scenario and see what others think.

Reply:
https://jira.mongodb.org/browse/SPEC-1511

Reply:
I have removed this restriction as seen in SPEC-1511.